### PR TITLE
add the request for /productDetails, remove unnecessary files and add mongoose

### PR DESCRIPTION
**Pull request by PanGan21**

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -14,10 +14,11 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@types/mongoose": "^5.5.17",
```
Comment (RogierIBM) [inline backend/package.json]: Please remove the ^

File: backend/package.json [hunk 2]
```diff
@@ -14,10 +14,11 @@
...
     "cors": "2.8.5",
     "express": "4.17.1",
     "mongodb": "3.3.2",
-    "mongoose": "5.7.0"
+    "mongoose": "^5.7.0"
```
Comment (RogierIBM) [inline backend/package.json]: Please remove the ^

File: backend/src/controllers/api.controller.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Request, Response, Router } from "express";
+
+const userSchema = require("../models/UserSchema");
+const productSchema = require("../models/ProductSchema");
+
+const apiController: Router = Router();
+
+apiController.get("/", (req: Request, res: Response): void => {
```
Comment (RogierIBM) [inline backend/src/controllers/api.controller.ts]: Is this endpoint still needed?

File: backend/src/controllers/api.controller.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  res.send(`You requested GET /users`);
+});
+
+apiController.post("/updateUser", (req: Request, res: Response): void => {
+  res.send("Thanks for the update man <3");
+});
+
+apiController.post("/", (req: Request, res: Response): void => {
```
Comment (RogierIBM) [inline backend/src/controllers/api.controller.ts]: Is this endpoint needed? Are we gonna create users through the frontend?

File: backend/src/models/ProductSchema.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import * as mongoose from "mongoose";
+
+const Schema = mongoose.Schema;
+
+const productSchema = new Schema({
+  name: {
+    type: String,
+    required: true
+  },
+  gronies: {
```
Comment (RogierIBM) [inline backend/src/models/ProductSchema.ts]: I think the correct spelling would be 'groenies'

File: backend/src/controllers/api.controller.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const user = new userSchema(
+    req.body.firstName,
+    req.body.lastName,
+    req.body.gronies
```
Comment (RogierIBM) [inline backend/src/controllers/api.controller.ts]: I believe the correct spelling is groenies

File: backend/src/models/UserSchema.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+const userSchema = new Schema({
+  firstName: {
+    type: String,
+    required: true
+  },
+  lastName: {
+    type: String,
+    required: true
+  },
+  gronies: {
```
Comment (RogierIBM) [inline backend/src/models/UserSchema.ts]: Should be groenies

File: backend/src/controllers/api.controller.ts (hunk 6 above)
Comment (PanGan21) [reply backend/src/controllers/api.controller.ts]: Are you sure?

**Pull request merged**